uxarray/_mpas.py

```diff
@@ -0,0 +1,266 @@
+import xarray as xr
+import numpy as np
+
+# remove/edit once unified fill value approach is implemented
+int_dtype = np.uint32
+fill_val = np.iinfo(int_dtype).max
+
+
+def _primal_to_ugrid(in_ds, out_ds):
+    """If input dataset is an unstructured MPAS file, function will encode the
+    MPAS Primal-Mesh in the UGRID conventions.
+
+    Parameters
+    ----------
+    in_ds : xarray.Dataset
+        Input MPAS dataset
+    out_ds : xarray.Dataset
+        Output dataset where the MPAS Primal-Mesh is encoded in the UGRID
+        conventions
+    """
+
+    # corners of primal-mesh cells (in degrees)
+    lonVertex = np.rad2deg(in_ds['lonVertex'].values)
+    latVertex = np.rad2deg(in_ds['latVertex'].values)
+
+    out_ds['Mesh2_node_x'] = xr.DataArray(
+        lonVertex,
+        dims=["nMesh2_node"],
+        attrs={
+            "standard_name": "longitude",
+            "long_name": "longitude of mesh nodes",
+            "units": "degrees_east",
+        })
+
+    out_ds['Mesh2_node_y'] = xr.DataArray(
+        latVertex,
+        dims=["nMesh2_node"],
+        attrs={
+            "standard_name": "latitude",
+            "long_name": "latitude of mesh nodes",
+            "units": "degrees_north",
+        })
+
+    # centers of primal-mesh cells (in degrees)
+    lonCell = np.rad2deg(in_ds['lonCell'].values)
+    latCell = np.rad2deg(in_ds['latCell'].values)
+
+    out_ds['Mesh2_face_x'] = xr.DataArray(
+        lonCell,
+        dims=["nMesh2_face"],
+        attrs={
+            "standard_name": "longitude",
+            "long_name": "longitude of center nodes",
+            "units": "degrees_east",
+        })
+
+    out_ds['Mesh2_face_y'] = xr.DataArray(
+        latCell,
+        dims=["nMesh2_face"],
+        attrs={
+            "standard_name": "latitude",
+            "long_name": "latitude of center nodes",
+            "units": "degrees_north",
+        })
+
+    # vertex indices that surround each primal-mesh cell
+    verticesOnCell = in_ds['verticesOnCell'].values
+
+    # max number of edges & vertices that make up each primal-mesh cell
+    nEdgesOnCell = in_ds['nEdgesOnCell'].values
+
+    # correct to be zero-indexed and to use fill values
+    verticesOnCell_fill = _add_fill_values(verticesOnCell, nEdgesOnCell)
+
+    out_ds["Mesh2_face_nodes"] = xr.DataArray(
+        data=verticesOnCell_fill,
+        dims=["nMesh2_face", "nMaxMesh2_face_nodes"],
+        attrs={
+            "cf_role": "face_node_connectivity",
+            "_FillValue": fill_val,
+            "start_index": int_dtype(0)
+        })
+
+    # vertex indices that saddle a given edge
+    verticesOnEdge = in_ds['verticesOnEdge'].values
+
+    # convert to zero-indexed
+    if verticesOnEdge.min() == 1:
+        verticesOnEdge -= 1
```

Comment:
I just noticed these kinds of checks.  These also aren't going to work for the same reason I pointed out -- zeros as invalid values.